zallet/src/components/chain_view.rs

```diff
@@ -0,0 +1,127 @@
+use std::fmt;
+use std::sync::Arc;
+
+use jsonrpsee::tracing::info;
+use tokio::{sync::RwLock, task::JoinHandle};
+use zaino_state::{
+    config::FetchServiceConfig,
+    fetch::{FetchService, FetchServiceSubscriber},
+    indexer::{IndexerService, IndexerSubscriber, ZcashService},
+    status::StatusType,
+};
+
+use crate::{
+    config::ZalletConfig,
+    error::{Error, ErrorKind},
+};
+
+#[derive(Clone)]
+pub(crate) struct ChainView {
+    // TODO: Migrate to `StateService`.
+    indexer: Arc<RwLock<Option<IndexerService<FetchService>>>>,
+}
+
+impl fmt::Debug for ChainView {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("ChainView").finish_non_exhaustive()
+    }
+}
+
+impl ChainView {
+    pub(crate) async fn new(
+        config: &ZalletConfig,
+    ) -> Result<(Self, JoinHandle<Result<(), Error>>), Error> {
+        let validator_rpc_address =
+            config
+                .indexer
+                .validator_listen_address
+                .unwrap_or_else(|| match config.network() {
+                    crate::network::Network::Consensus(
+                        zcash_protocol::consensus::Network::MainNetwork,
+                    ) => "127.0.0.1:8232".parse().unwrap(),
+                    _ => "127.0.0.1:18232".parse().unwrap(),
+                });
+
+        let db_path = config
+            .indexer
+            .db_path
+            .clone()
+            .ok_or(ErrorKind::Init.context("indexer.db_path must be set (for now)"))?;
+
+        let config = FetchServiceConfig::new(
+            validator_rpc_address,
+            config.indexer.validator_cookie_auth.unwrap_or(false),
+            config.indexer.validator_cookie_path.clone(),
+            config.indexer.validator_user.clone(),
+            config.indexer.validator_password.clone(),
+            None,
+            None,
+            config.indexer.map_capacity,
+            config.indexer.map_shard_amount,
+            db_path,
+            config.indexer.db_size,
+            config.network().to_zebra(),
+            false,
+            // Setting this to `true` causes start-up to block on completely filling the
+            // cache. Zaino's DB currently only contains a cache of CompactBlocks, so we
+            // make do for now with uncached queries.
+            // TODO: https://github.com/zingolabs/zaino/issues/249
+            true,
```

Comment:
"We make do for now with uncached queries" would imply to me that this is being set to false to avoid the performance problem on startup, but it is set to true, so the comment doesn't seem to match the code.

Reply:
The comment is incorrect here, not the code.